### ISSUE TITLE
"make" fails

### ISSUE THREAD

**Issue by derandre24** [gschorcht/i2c-ch341-usb#6]
Hello everyone, 

when I try to compile the driver i get an error.
It looks like there is maybe a problem with the kernel headers.

I did the following.

sudo apt-get update
sudo apt-get upgrade
sudo apt-get install dkms
git clone https://github.com/gschorcht/i2c-ch341-usb.git
cd i2c-ch341-usb
make

thomas@searay:~/i2c-ch341-usb$ make
make -C /lib/modules/5.13.0-22-generic/build M=/home/thomas/i2c-ch341-usb  modules
make[1]: Entering directory '/usr/src/linux-headers-5.13.0-22-generic'
  MODPOST /home/thomas/i2c-ch341-usb/Module.symvers
ERROR: modpost: "irq_to_desc" [/home/thomas/i2c-ch341-usb/i2c-ch341-usb.ko] undefined!
make[2]: *** [scripts/Makefile.modpost:150: /home/thomas/i2c-ch341-usb/Module.symvers] Error 1
make[2]: *** Deleting file '/home/thomas/i2c-ch341-usb/Module.symvers'
make[1]: *** [Makefile:1794: modules] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.13.0-22-generic'
make: *** [Makefile:20: i2c-ch341-usb.ko] Error 2

**Comment by E011011101001** [comment]
It's the same problem as mentioned in [spi-ch341-usb#18](https://github.com/gschorcht/spi-ch341-usb/issues/18#issue-858952651).

**Comment by antoninhrlt** [comment]
I've the same problem, have you found a solution ?

**Comment by derandre24** [comment]
I am now using a Linux distribution with a 4.xx kernel.

**Comment by E011011101001** [comment]
> I've the same problem, have you found a solution ?

@antoninhrlt I changed the kernel source `irqdesc.c` and exported the `irq_to_desc` manually as a workaround.

**Comment by sujiwo** [comment]
> @antoninhrlt I changed the kernel source `irqdesc.c` and exported the `irq_to_desc` manually as a workaround.
Could you please show the changes? Thank you.

**Comment by E011011101001** [comment]
I removed the macro wrapping [this line](https://github.com/torvalds/linux/blob/9f7fb8de5d9bac17b6392a14af40baf555d9129b/kernel/irq/irqdesc.c#L356) and built the changed kernel. Now I boot that kernel whenever I need to use this driver.

(BTW, I don't know much about kernel development, so my workaround could cause some damages I don't understand.)

**Comment by xboxplayer9889** [comment]
What do you think about changing line 660 of i2c-ch341-usb.c to
ch341_dev->irq_descs[i] = irq_data_to_desc( irq_get_irq_data(ch341_dev->irq_base + i) );

I think it should work...

**Comment by sujiwo** [comment]
> What do you think about changing line 660 of i2c-ch341-usb.c to ch341_dev->irq_descs[i] = irq_data_to_desc( irq_get_irq_data(ch341_dev->irq_base + i) );
> 
> I think it should work...

I can confirm this solution works
(Ubuntu 20.04, kernel 5.13.0, using MPU6050).

**Comment by antoninhrlt** [comment]
I have just another problem but now, `irq_to_desc` is not longer a problem !
> My problem : \
> `Skipping BTF generation for /mnt/Data/Documents/i2c-ch341-usb/i2c-ch341-usb.ko due to unavailability of vmlinux`

**Comment by koalabi** [comment]
I have tried xboxplayer9889's proposed fix and it seems indeed to work. Why isn't the issue closed then?

**Comment by antoninhrlt** [comment]
> I have tried xboxplayer9889's proposed fix and it seems indeed to work. Why isn't the issue closed then?

Yes i don't know why my pull request fixing this problem was even not merged/closed

It seems repository's owner does not look at his project often

**Comment by BlackMage2** [comment]
hmpf last change from gschorcht was on Oct 22, 2021

and now is Sep 2023
